includes/compat/class-wc-stripe-subscriptions-legacy-sepa-token-update.php

```diff
@@ -100,85 +107,54 @@ private function get_subscription_to_migrate( $subscription_id ) {
 			throw new \Exception( sprintf( '---- Skipping migration of subscription #%d. Subscription not found.', $subscription_id ) );
 		}
 
-		if ( WC_Gateway_Stripe_Sepa::ID !== $subscription->get_payment_method() ) {
-			throw new \Exception( sprintf( '---- Skipping migration of subscription #%d. Subscription is not using the legacy SEPA payment method.', $subscription_id ) );
-		}
-
 		return $subscription;
 	}
 
 	/**
-	 * Returns an updated token to be used for the subscription, given the source ID.
+	 * Updates the payment method used for renewals to the migrated pm_, if any.
 	 *
-	 * If no updated token is found, we create a new one based on the legacy one.
+	 * The subscription is using a source for renewals at this point.
+	 * When the migration runs on the Stripe account, there will be a payment method (pm_) migrated from the source (src_).
+	 * This method updates the subscription to use the migrated payment method (pm_) for renewals, if it exists.
 	 *
-	 * @param string  $source_id The Source or Payment Method ID associated with the subscription.
-	 * @param integer $user_id   The WordPress User ID to whom the subscription belongs.
+	 * @param WC_Subscription $subscription The subscription to update.
+	 * @throws \Exception When the subscription is already using a pm_ or its src_ hasn't been migrated to a pm_.
 	 */
-	private function maybe_create_updated_sepa_token_by_source_id( string $source_id, int $user_id ) {
-
-		// Retrieve the updated SEPA tokens for the user.
-		$replacement_token = $this->get_customer_token_by_source_id( $source_id, $user_id, $this->updated_sepa_gateway_id );
-
-		// If no updated SEPA token was found, create a new one based on the source ID.
-		if ( ! $replacement_token ) {
-			$replacement_token = $this->create_updated_sepa_token( $source_id, $user_id );
-		}
-	}
-
-	/**
-	 * Get the token for the user by its source ID and gateway ID. s
-	 *
-	 * @param string  $source_id  The ID of the source we're looking for.
-	 * @param integer $user_id    The ID of the user we're retrieving tokens for.
-	 * @param string  $gateway_id The ID of the gateway of the tokens we want to check.
-	 *
-	 * @return WC_Payment_Token|false
-	 */
-	private function get_customer_token_by_source_id( string $source_id, int $user_id, string $gateway_id ) {
-		$customer_tokens = WC_Payment_Tokens::get_customer_tokens( $user_id, $gateway_id );
+	private function set_subscription_updated_payment_method( WC_Subscription $subscription ) {
+		$source_id = $subscription->get_meta( self::SOURCE_ID_META_KEY );
 
-		foreach ( $customer_tokens as $token ) {
-			if ( $source_id === $token->get_token() ) {
-				return $token;
-			}
+		// Bail out if the subscription is already using a pm_.
+		if ( ! str_starts_with( $source_id, 'src_' ) ) {
```

Comment:
```suggestion
		if ( 0 === strpos( $source_id, 'src_' ) ) {
```

`str_starts_with` was added in php 8.0 and WC currently allows back to PHP 7.4.

Reply:
Whoa, that's a great catch!! Completely missed it. Thanks!